docs_nnx/guides/pytree.md

```diff
@@ -196,64 +306,119 @@ foo = Foo()
 
 ```{code-cell} ipython3
-class Block(nnx.Module):
-  def __init__(self, din: int, dout: int, rngs: nnx.Rngs):
-    self.mode = 1
-    self.linear = nnx.Linear(din, dout, rngs=rngs)
-    self.bn = nnx.BatchNorm(dout, rngs=rngs)
-    self.dropout = nnx.Dropout(0.1, rngs=rngs)
+class Shared(nnx.Pytree):
+  def __init__(self):
+    self.x = jnp.array(1.0)
 
-  def __call__(self, x):
-    return nnx.relu(self.dropout(self.bn(self.linear(x))))
-  
-model = Block(din=1, dout=2, rngs=nnx.Rngs(0))
-
-print("train:")
-print(f"- {model.mode = }")
-print(f"- {model.bn.use_running_average = }")
-print(f"- {model.dropout.deterministic = }")
-
-# Set attributes for evaluation
-model.set_attributes(deterministic=True, use_running_average=True, mode=2)
-
-print("eval:")
-print(f"- {model.mode = }")
-print(f"- {model.bn.use_running_average = }")
-print(f"- {model.dropout.deterministic = }")
+class Parent(nnx.Pytree):
+  def __init__(self):
+    self.left = Shared()
+    self.right = self.left  # reference sharing
+
+m = Parent()
+
+print(f"{nnx.find_duplicates(m) = }  # not a tree")
+```
+
+The main issue is that sharing is not preserved across pytree operations including JAX transforms, and this results in unintended state duplication:
+
+```{code-cell} ipython3
+m = Parent()
+print(f"Before: {m.left is m.right = }")
+
+@jax.jit
+def f(m):
+  print(f"Inside: {m.left is m.right = }")
+  return m
+
+m = f(m)
+print(f"After:  {m.left is m.right = }")
+```
+
+Reference sharing is rare in most Machine Learning applications, however if it is required you can either use the `nnx.{split, merge, state, update}` APIs to move the deduplicated state and graph definiton across the JAX transforms:
+
+```{code-cell} ipython3
+m = Parent()
+print(f"Before: {m.left is m.right = }")
+graphdef, state = nnx.split(m)
+
+@jax.jit
+def f(graphdef, state):
+  m = nnx.merge(graphdef, state)
+  print(f"Inside: {m.left is m.right = }")
+  return nnx.state(m)
+
+state = f(graphdef, state)
+nnx.update(m, state)
+
+print(f"After:  {m.left is m.right = }")
+print(f"{state = }") # deduplicated state
 ```
 
+Or alternatively you can use the NNX transforms which preserve shared references:
+
 ```{code-cell} ipython3
-model = Block(din=1, dout=2, rngs=nnx.Rngs(0))
-
-model.eval(mode=2)  # .set_attributes(deterministic=True, use_running_average=True, mode=2)
-print("eval:")
-print(f"- {model.mode = }")
-print(f"- {model.bn.use_running_average = }")
-print(f"- {model.dropout.deterministic = }")
-
-model.train(mode=1)  # .set_attributes(deterministic=False, use_running_average=False, mode=1)
-print("train:")
-print(f"- {model.mode = }")
-print(f"- {model.bn.use_running_average = }")
-print(f"- {model.dropout.deterministic = }")
+m = Parent()
+print(f"Before: {m.left is m.right = }")
+
+@nnx.jit
+def f(m):
+  print(f"Inside: {m.left is m.right = }")
+  return m
+
+m = f(m)
+
+print(f"After:  {m.left is m.right = }")
+```
+
+### pytree=False
```

Comment:
Let's call the title "Backward compatibility" or "Turning off pytree registration"